cli/cache/emit.rs

```diff
@@ -1,71 +1,206 @@
 // Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
 
+use std::path::PathBuf;
+
 use deno_ast::ModuleSpecifier;
+use deno_core::anyhow::anyhow;
 use deno_core::error::AnyError;
+use deno_core::serde_json;
+use serde::Deserialize;
+use serde::Serialize;
 
-use super::CacheType;
-use super::Cacher;
+use super::DiskCache;
+use super::FastInsecureHasher;
 
-/// Emit cache for a single file.
-#[derive(Debug, Clone, PartialEq)]
-pub struct SpecifierEmitCacheData {
+#[derive(Debug, Deserialize, Serialize)]
+struct EmitMetadata {
   pub source_hash: String,
```

Comment:
This changed from `version_hash` (diff is not clear because I moved this struct here). It's using a faster xxHash now so worth it to rename I think.

Also, if someone upgrades deno, uses it a bit, then downgrades, older versions of deno reading this file will re-emit.

Reply:
Add as a docstring that Deno version mismatch will cause to skip the cache?